### ISSUE TITLE
Getting NewGameState but can't compare its values.

### ISSUE THREAD

**Issue by joao7yt** [rakijah/CSGSI#25]
I'm testing things out so I wrote this snippet. I get a lot of "New game state." outputs in the console, meaning I'm sure getting updates in the game state, but it doesn't output "Bomb is planted." when it is planted. Do you know why? I also tried using the events you provide with CSGSI but it doesn't work as well.

```
using System;
using System.Threading.Tasks;
using System.Net.Http;
using CSGSI;
using CSGSI.Nodes;

namespace Test
{
    class Program
    {
        static GameStateListener gsl;
        static void Main(string[] args)
        {
            gsl = new GameStateListener(3000);
            gsl.NewGameState += new NewGameStateHandler(OnNewGameState);
            if (!gsl.Start())
            {
                Environment.Exit(0);
            }
            Console.WriteLine("Listening...");
        }

        static void OnNewGameState(GameState gs)
        {
            Console.WriteLine("New game state.");

            // EDIT:
            // Console.WriteLine(gs.Bomb.State.ToString());

            // EDIT 2:
            // Console.WriteLine(gs.JSON.ToString());

            if (gs.Bomb.State == BombState.Planted)
            {
                Console.WriteLine("Bomb is planted.");
            }
        }
    }
}
```

EDIT: So, I added "Console.WriteLine(gs.Bomb.State.ToString());" in "OnNewGameState(GameState gs)" and the output is always "Undefined"... So I think there's a problem with CSGSI, right?

EDIT 2: Using "Console.WriteLine(gs.JSON.ToString());" I could see that "bomb": "planted" is being added to "round", but for some reason CSGSI is not setting Bomb.State to planted...

**Comment by YorVeX** [comment]
I wonder whether we are using the same CSGSI library? Because I don't even have a "gs.Bomb.State" field, I get the bomb status from "gs.Round.Bomb" and that works perfectly fine.

Except for the fact that you never get that event at the exact time when the bomb is planted. That's intentional by Valve, because when the CSGSI was new people used it to write bomb timer tools so they could see whether you can still defuse or not. To counter this they added some randomness to the event, it is always fired a few seconds before or after the actual bomb plant.

**Comment by joao7yt** [comment]
Well, check at the main page of CSGSI: https://github.com/rakijah/CSGSI

At the bottom there's some example codes... I just used it...

I got my version of CSGSI from NuGet Package Manager in the Visual Studio 2017.

**Comment by joao7yt** [comment]
In fact when trying to use gs.Round.Bomb it says the round node from the library doesn't even have a definition for Bomb:

> Error	CS1061	'RoundNode' does not contain a definition for 'Bomb' and no accessible extension method 'Bomb' accepting a first argument of type 'RoundNode' could be found (are you missing a using directive or an assembly reference?)

**Comment by joao7yt** [comment]
Seems that the 1.3.0 version of CSGSI doesn't contain gs.Round.Bomb, but version 1.2.2 does. I'm gonna test with the previous version to see if it works.

**Comment by joao7yt** [comment]
That's right, it does work now with version 1.2.2, so there's a bug with version 1.3.0.

**Comment by YorVeX** [comment]
Right, that would be the explanation. Since my version was from December I assumed I would have the latest one but now I remember that I am using a self-compiled version and changed it myself in December to support Danger Zone. Shortly after that an update was released that includes those changes and others but I never remembered to update again (especially since I am happy with how everything currently works, it's super stable and got everything I need).

**Comment by rakijah** [comment]
Hey, I'm sorry it took so long for me to respond.

I've looked at your report and did some testing myself. 

It seems the `gs.Bomb` node is not sent when you're playing in a match yourself (only when spectating). Either I have misinterpreted this when I took out `gs.Round.Bomb` or Valve themselves changed the way this works, as I am fairly certain I tested this node both in offline matches and while viewing a demo. The information is now split into `gs.Bomb` (only available while spectating) and `gs.Round.Bomb` (always available).

I have just submitted a commit and a new version for the NuGet package that re-adds the `gs.Round.Bomb` (and also implements `gs.Bomb.Countdown` that was missing).

Hopefully this fixes this issue for you; if not, feel free to contact me.
